### DISCUSSION TITLE
Recommended CDN with "Accept" Cache Key

### DISCUSSION THREAD

**Discussion by mstaack** [imgproxy/imgproxy#1261] [Q&A] [answered]
Hey there,

i am in the progress of setting up our imgproxy solution.

docs mention (https://docs.imgproxy.net/configuration/options#avifwebp-support-detection):

> When AVIF/WebP support detection is enabled, please take care to configure your CDN or caching proxy to take the Accept HTTP header into account while caching.

any recommended cdn providers which support that?

looked at:
- cloudflare.com does only support custom cache keys in their enterprise (yeah right, contact sales pricing) plan
https://developers.cloudflare.com/cache/how-to/cache-keys/
- bunny.net cdn their cache key "vary" does not support the accept header

or what is the best way to approach cdn caching?

**Comment by mstaack** [comment]
okay `bunny.net` does support vary header for webp/avif if i read this correctly:
https://support.bunny.net/hc/en-us/articles/360012186820-How-does-Vary-Cache-work

**Comment by mstaack** [comment]
also cloudflare seems to support `vary` header for images in the pro plan:
https://developers.cloudflare.com/cache/advanced-configuration/vary-for-images/

**Comment by DarthSim** [comment]
We usually recommend AWS CloudFront since it allows pretty precise tuning caching behavior.

Both Clouflare's and Bunny's solutions look like what you're looking for, but they have URL extension limitations.

**Reply by mstaack** [reply]
what exactly do you mean by url extension limitations? plan was to use bunny.net

**Reply by DarthSim** [reply, marked as answer]
From Bunny's docs:

> This key is only applied for URLs with the following extensions: jpg, jpeg, webp, png, gif.
